sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -156,6 +156,33 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       structDf.select(xxhash64($"a", $"record.*")))
   }
 
+  test("SPARK-28224: Aggregate sum big decimal overflow") {
+    Seq(true, false).foreach { nullOnOverflow =>
+      withSQLConf((SQLConf.DECIMAL_OPERATIONS_NULL_ON_OVERFLOW.key, nullOnOverflow.toString)) {
+        val structDf = largeDecimals.select("a").agg(sum("a"))
+        if (nullOnOverflow) {
+          checkAnswer(structDf, Row(null))
+        } else {
+          val e = intercept[Exception] {
+            structDf.collect
+          }
+          assert(e.getCause.getClass.equals(classOf[ArithmeticException]))
+          assert(e.getCause.getMessage.contains("cannot be represented as Decimal"))
+        }
+      }
+    }
+  }
+
+  test("Aggregate sum integers") {
```

Comment:
We need the two tests below for this change?

Reply:
@maropu the change introduces branching for BigDecimal case, so I want to verify that sum still works for integer/double case.

Reply:
But, the case for int/double is an existing path in master, so we already have tests for that somewhere, right? I this the test above is enough for this pr.